ArdusatSDK.cpp

```diff
@@ -453,266 +257,1212 @@ const char * valuesToCSV(const char *sensorName, unsigned long timestamp, int nu
-
-  dtostrf(input.t, 2, 3, _getOutBuf() + _output_buf_len);
-  _output_buf_len = strlen(_getOutBuf());
-
-  _add_checksum_to_csv_buffer(sensorName, 1, input.t);
-  _getOutBuf()[_output_buf_len++] = '\n';
 
-  return _getOutBuf();
+/**************************************************************************//**
+ * @brief   Constructs Acceleration sensor object
+ * @ingroup acceleration
+ *
+ * Example Usage:
+ * @code
+ *     Acceleration accel = Acceleration(); // Instantiate sensor object
+ *     accel.begin();                       // Initialize sensor
+ *     Serial.println(accel.readToJSON());  // Read and print values in JSON
+ * @endcode
+ *****************************************************************************/
+Acceleration::Acceleration(void) {
+  Acceleration::sensorId = SENSORID_ADAFRUIT9DOFIMU;
+  Acceleration::gGain = LSM303_ACCEL_GAIN8G;
+  Acceleration::header.unit = DATA_UNIT_METER_PER_SECONDSQUARED;
+  Acceleration::initialized = false;
 }
 
-_DEFAULT_SENSOR_NAME_TO_CSV_FXN(gyro)
-const char * gyroToCSV(const char *sensorName, gyro_t & input) {
-  _headerToCSV(&(input.header), sensorName);// warning : resets the internal buffer
-
-  dtostrf(input.x, 2, 3, _getOutBuf() + _output_buf_len);
-  _output_buf_len = strlen(_getOutBuf());
-  _getOutBuf()[_output_buf_len++] = CSV_SEPARATOR;
+Acceleration::~Acceleration() {
+}
 
-  dtostrf(input.y, 2, 3, _getOutBuf() + _output_buf_len);
-  _output_buf_len = strlen(_getOutBuf());
-  _getOutBuf()[_output_buf_len++] = CSV_SEPARATOR;
+/**
+ * @brief   Initializes the sensor with basic configurations
+ * @ingroup acceleration
+ *
+ * @retval true  Successfully initialized
+ * @retval false Failed to initialize
+ */
+boolean Acceleration::begin(void) {
+  return Acceleration::begin(Acceleration::gGain);
+}
 
-  dtostrf(input.z, 2, 3, _getOutBuf() + _output_buf_len);
-  _output_buf_len = strlen(_getOutBuf());
+/**
+ * @brief   Initializes the sensor with advanced configurations
+ * @ingroup acceleration
+ *
+ * @param gGain Advanced configuration for accelerometer's gain
+ *     - LSM303_ACCEL_GAIN2G
+ *     - LSM303_ACCEL_GAIN4G
+ *     - LSM303_ACCEL_GAIN6G
+ *     - LSM303_ACCEL_GAIN8G (Default)
+ *     - LSM303_ACCEL_GAIN16G
+ *
+ * @retval true  Successfully initialized
+ * @retval false Failed to initialize
+ */
+boolean Acceleration::begin(lsm303_accel_gain_e gain) {
+  catchSpaceboard();
+  Acceleration::gGain = gain;
+  Acceleration::initialized = start_sensor_or_err(acceleration_sensor_name, lsm303_accel_init(gain));
+  return Acceleration::initialized;
+}
 
-  _add_checksum_to_csv_buffer(sensorName, 3, input.x, input.y, input.z);
-  _getOutBuf()[_output_buf_len++] = '\n';
+/**
+ * @brief   Takes a reading from the sensor
+ * @ingroup acceleration
+ */
+void Acceleration::read(void) {
+  Acceleration::header.timestamp = millis();
 
-  return _getOutBuf();
+  Acceleration::header.sensor_id = Acceleration::sensorId;
```

Comment:
I would either define a `Sensor::read` (for the ABC) that gets called by each `read` function, or define a `Sensor::writeHeader` function that everyone can access, just so we don't have to write this sort of logic everywhere.